src/plot.R

```diff
@@ -1,41 +1,92 @@
     geom_line() +
     theme_minimal() +
-    labs(y = "CPUtilization (%)", x = "Time (min)")
+    labs(y = "CPUtilization (%)", x = "Time (hour)", color = "") +
+    theme(legend.position = "top", legend.direction = "horizontal") +
+    scale_x_datetime(date_breaks = date_break, date_labels = "%H:%M")
   
-  if (policy_name == "simple_scaling") {
+  if (startsWith(title, "Simple Scaling")) {
+    # Draw a line for each threshold
     upper_bound <- policy_parameters$upper_bound
-    lower_bound <-policy_parameters$lower_bound
+    lower_bound <- policy_parameters$lower_bound
     
     plot <- plot +
       geom_hline(yintercept = upper_bound, color = "red") +
-      geom_text(aes(x = 70, y = upper_bound+2, label = "Upper Bound"), size = 3, color = "red") +
+      geom_text(aes(x = min(datetime),y = upper_bound+2, label = "Upper Bound"), size = 2, color = "red") +
       geom_hline(yintercept = lower_bound, color = "blue") +
-      geom_text(aes(x = 70, y = lower_bound-2, label = "Lower Bound"), size = 3, color = "blue")
-  
-  } else if (policy_name == "target_tracking") {
+      geom_text(aes(x = min(datetime), y = lower_bound-2, label = "Lower Bound"), size = 2, color = "blue")
+
+  } else if (startsWith(title, "Target Tracking")) {
+    # Draw a line for target threshold
     target <- policy_parameters$target_value
     
     plot <- plot +
       geom_hline(yintercept = target, color = "red") +
       geom_text(aes(x = 120, y = target+2, label = "Target threshold"), size = 3, color = "red")
-  
+    
   }
+  
+  ggplot2::ggsave(filepath, width = 7, height = 4)
+}
 
-  utilization_plot_filepath <- paste("output/", policy_name, "_utilization.pdf", sep='')
-  ggplot2::ggsave(utilization_plot_filepath, width = 7, height = 4)
+plot_cores <- function(data, title, date_break, filepath) {
+  # Add plot both real and simulation
+  data <- data %>%
+    rename(simulation = AllocatedCores, real = RealAllocatedCores) %>%
+    pivot_longer(c(simulation, real))
   
-  data %>% ggplot(aes(x = timestamp - min(timestamp), y = AllocatedCores)) +
+  # Plot allocated cores over the time for both real and simulation
+  data %>% ggplot(aes(x = datetime, y = value, color = name)) +
+    ggtitle(title) +
     geom_line() +
     theme_minimal() +
-    labs(y = "Cores", x = "Tempo (min)")
+    labs(y = "Cores", x = "Time (hour)", color = "") +
+    theme(legend.position = "top", legend.direction = "horizontal") +
+    scale_x_datetime(date_breaks = date_break, date_labels = "%H:%M")
   
-  cores_plot_filepath <- paste("output/", policy_name, "_cores.pdf", sep='')
-  ggplot2::ggsave(cores_plot_filepath, width = 7, height = 4)
+  ggplot2::ggsave(filepath, width = 7, height = 4)
+}
+
+plot_simulation <- function(data, policy_parameters, configs) {
+  # Creates two plots for cores and system utilization over time comparing
+  # real and simulation results.
+  title <- get_title(data, configs$policies$use)
+  date_break <- get_plot_date_break(configs)
```

Comment:
```suggestion
  date_break <- ifelse(is.null(configs$plot_scale_datetime), "1 hour", configs$plot_scale_datetime)
```

Acho meio desnecessário criar uma função que faz apenas um if...else.